devito/types/dense.py

```diff
@@ -1077,32 +1058,60 @@ def _eval_at(self, func):
             return self.subs(mapper)
         return self
 
+    @classmethod
+    def __staggered_setup__(cls, dimensions, **kwargs):
+        """
+        Setup staggering-related metadata. This method assigns:
+
+            * 0 to non-staggered dimensions;
+            * 1 to staggered dimensions.
+        """
+        stagg = kwargs.get('staggered')
```

Comment:
nitpicking: I'd call this value `staggered`, and the other one (the "lowered" staggered" `processed` instead, or perhaps just a letter, but I would avoid both `stagg` and `staggered`